pkg/interceptors/interceptors.go

```diff
@@ -171,28 +171,40 @@ func UnmarshalParams(ip map[string]interface{}, p interface{}) error {
 	return nil
 }
 
-// ResolveURL returns the URL for the given core interceptor
-func ResolveURL(i *triggersv1.TriggerInterceptor) *url.URL {
+// Getname returns the name for the given core interceptor
+func GetName(i *triggersv1.TriggerInterceptor) string {
 	// This is temporary until we implement #868
-	path := ""
+	name := ""
 	switch {
 	case i.Bitbucket != nil:
-		path = "bitbucket"
+		name = "bitbucket"
 	case i.CEL != nil:
-		path = "cel"
+		name = "cel"
 	case i.GitHub != nil:
-		path = "github"
+		name = "github"
 	case i.GitLab != nil:
-		path = "gitlab"
+		name = "gitlab"
 	}
-	return &url.URL{
-		Scheme: "http",
-		Host:   fmt.Sprintf("%s.%s.svc", CoreInterceptorsHost, os.Getenv("TEKTON_INSTALL_NAMESPACE")),
-		Path:   path,
+	return name
+}
+
+// Resolve resolves the given CEL name to the
+type InterceptorGetter func(name string) (*triggersv1.Interceptor, error)
+
+var ErrNoURL = errors.New("interceptor URL was not found")
```

Comment:
I wonder if this shouldn't be an error type, so that we can have the URL that was not found in the message to help with debugging?

Reply:
oh we return this when the URL field is nil...so we literally have nothing to return. I can see this being confusing. I'll change the error message to URL was nil or something like that